### ISSUE TITLE
Cannot log into MinIO GUI with LDAP STS Temporary Credentials

### ISSUE THREAD

**Issue by jdheyburn** [minio/minio#9836]
## Expected Behavior
If using [LDAP Auth](https://github.com/minio/minio/blob/master/docs/sts/ldap.md), the temporary credentials obtained from STS following successful LDAP authentication should then allow the user to log into the GUI with the provided access and secret key.

## Current Behavior
When the obtained access and secret keys are using in the GUI, an error message is returned indicating the credentials are not valid.

## Possible Solution
N/A

## Steps to Reproduce (for bugs)

Setup is similar to #9593.

1. Following setup in #9593, make a call to the [AssumeRoleWithLDAPIdentity](https://github.com/minio/minio/blob/master/docs/sts/ldap.md#sample-post-request) Endpoint 
  - Ensure that the `MINIO_IDENTITY_LDAP_STS_EXPIRY` field is set to something high (1h e.g.)
2. In the response to 1., extract the Access Key and Secret Key and use them in the MinIO GUI
3. The credentials will fail

## Context
We are using minio in gateway s3 mode, and would like to allow users to log into MinIO using their Active Directory credentials.

## Regression
I am not sure if we are misinterpreting the purpose of LDAP here.

## Your Environment

* Version used (`minio version`): `RELEASE.2020-06-14T18-32-17Z`
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

**Comment by harshavardhana** [comment]
STS creds are meant for API access, by applications which are aware of temporary access 

UI is an anti-pattern for this that's why we wrote mcs 

This is on purpose avoided you should look at our

https://github.com/minio/mcs

This shall primarily replace all the UI related functionality, PTAL.

**Comment by jdheyburn** [comment]
This looks very cool @harshavardhana. Can you let me know if the below diagram indicates where mcs fits in the rest of the stack?

```
User -> MCS -> MinIO Server (Gateway S3 Mode) -> AWS S3
                          ↓
                        etcd
```

I just tried the above, but was not able to log into mcs with a local user and password, while I could use the same credentials for the MinIO server. I raised an issue at that repo for this error-  https://github.com/minio/mcs/issues/176

I also tried the same with MinIO server with LDAP Auth but was not able to get it to work. Do you have an example of implementation? 

Thanks for your time.

**Comment by harshavardhana** [comment]
> User -> MCS -> MinIO Server (Gateway S3 Mode) -> AWS S3
>                           ↓
>                         etcd

This is the correct stack @jdheyburn

**Comment by harshavardhana** [comment]
> I just tried the above, but was not able to log into mcs with a local user and password, while I could use the same credentials for the MinIO server. I raised an issue at that repo for this error- [minio/mcs#176](https://github.com/minio/mcs/issues/176)
> 
> I also tried the same with MinIO server with LDAP Auth but was not able to get it to work. Do you have an example of implementation?

It doesn't support gateway yet, please use the MinIO server mode - gateway is not our primary focus - it was primarily implemented for users to migrate to MinIO eventually. 

We are debating to support SessionToken on the UI as well we will keep you posted.

**Comment by jdheyburn** [comment]
I understand gateway isn't heavily supported so I appreciate the time yourself and the team are putting into it. Perhaps it may help if I explained how we are using Minio.

Our use case is providing proxied access to S3 so that we can have control over what data leaves AWS and our VPC. Minio is in a gateway s3 mode that is only accessible from an EC2 instance in a private subnet. We've blocked all other AWS console access from this instance to prevent data from leaving the environment (i.e. a user could log into AWS with their personal account and upload data to their S3 bucket).

Currently we have Minio set up with local credentials for users, however we would ideally like LDAP Auth against our Active Directory domain, so that users only have 1 set of credentials to remember. Also allowing us to provide access depending on what AD group they are in.
